django/thunderstore/core/utils.py

```diff
@@ -70,14 +70,37 @@ def capture_exception(*args, **kwargs):
     capture_sentry_exception(*args, **kwargs)
 
 
-def make_full_url(request: Optional[HttpRequest], path: str):
+def enforce_url_protocol(
+    url: str,
+):
+    if settings.PROTOCOL == "https://" and url.startswith("http://"):
+        url = f"https://{url[7:]}"
+    return url
+
+
+def build_url_from_request_object(
+    path: str,
+    request: Optional[HttpRequest],
+):
     """Build an URL relative to a request using the proper request scheme"""
     url = path
     if request:
         url = request.build_absolute_uri(url)
-    if settings.PROTOCOL == "https://" and url.startswith("http://"):
-        url = f"https://{url[7:]}"
-    return url
+    return enforce_url_protocol(url)
+
+
+def build_url_from_site_object(
+    path: str,
+    site=None,
+):
+    if site is None:
+        raise Exception("Missing site object")
```

Comment:
Why does this allow (and default to) `None` in the first place if we're just going to raise an exception?

Also bare `Exception` usage is generally a bad idea, use `AttributeError` at least.

Reply:
This probably skipped my mind whilst doing this. Nice catch, I'll edit it 👌